### PR TITLE
chore: update image to gcr.io/k8s-staging-node-ipam-controller and registry.k8s.io/node-ipam-controller.

### PR DESCRIPTION
**Pull request by liangyuanpeng**
* add cloudbuild file.

* update docker-build to image-build, consistency with other sig projects,like kueue.

Basically use PR https://github.com/kubernetes-sigs/lws/pull/42 as a reference for the modifications.

part of #15

### CODE REVIEW

File: cloudbuild.yaml
Comment (liangyuanpeng) [inline cloudbuild.yaml:1]: This file will use by test-infra ,  https://github.com/kubernetes/test-infra/pull/32373

**Review by liangyuanpeng — COMMENTED** [review]
/assign @aojea
/assign @ameukam

File: cloudbuild.yaml
Comment (aojea) [reply cloudbuild.yaml:1]: you need to follow these steps https://github.com/kubernetes/test-infra/tree/4c1fb88972509493792e8f9c51feccbf49518503/config/jobs/image-pushing#image-pushing-jobs

**Comment by aojea** [comment]
/lgtm
/approve

@liangyuanpeng apologies for not being able to follow up, are you still willing to work on this?

**Pull request merged**